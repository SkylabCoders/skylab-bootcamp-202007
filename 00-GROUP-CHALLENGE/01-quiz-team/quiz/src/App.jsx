import React from 'react';
import './css/app.scss';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import PageNotFound from './components/PageNotFound';
import Header from './components/Header';
import Footer from './components/Footer';
import LeaderBoard from './components/LeaderBoard';
import LoginScreen from './components/LoginScreen';
import Dashboard from './components/Dashboard';
import Question from './components/Question';
import Game from './components/Game';
import Answer from './components/Answer';
import * as ROUTES from './config/routes';

function App() {
	return (
		<div className="app">
			<Router>
			<Header />
			<Switch>
				<Route path={ROUTES.HOME} exact component={Dashboard} />
				<Route path={ROUTES.LEADERBOARD} component={LeaderBoard} />
				<Route path={ROUTES.LOGIN} component={LoginScreen} />
				<Route path={ROUTES.QUESTION} component={Question} />
				<Route path={ROUTES.THEMES_WILDCARD} component={Game} />
				<Route path={ROUTES.ANSWER} component={Answer} />
				<Route component={PageNotFound} />
			</Switch>
			<Footer />
			</Router>
		</div>
	);
}

export default App;
