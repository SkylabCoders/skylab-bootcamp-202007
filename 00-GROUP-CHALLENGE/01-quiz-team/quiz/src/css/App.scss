.app {
  width: 100%;
}

/*# sourceMappingURL=App.scss.map */
