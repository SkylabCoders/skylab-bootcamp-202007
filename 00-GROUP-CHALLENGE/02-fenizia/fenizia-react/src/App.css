* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Lato', sans-serif;
}


.margin-body {
  padding-top: 60px;
  min-height: calc(100vh - 140px - 0px); 
}
