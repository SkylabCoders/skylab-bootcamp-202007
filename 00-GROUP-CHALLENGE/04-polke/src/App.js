import React from 'react';
import HeaderNavbar from './components/Navbar/HeaderNavbar';
import Footer from './components/Footer/Footer';
import Landing from './components/Landing/Landing';
import RepoDetail from './components/RepoDetail/RepoDetail';
import UserDetail from './components/UserDetail/UserDetail';
import AuthComp from './components/AuthComp/AuthComp';
import { Route, Switch } from 'react-router-dom';

function App() {
	return (
		<>
			<HeaderNavbar />
			<Switch>
				<Route path="/" exact component={Landing} />
				<Route path="/userDetail" exact component={UserDetail} />
				<Route path="/repoDetail/:userName/:repoName" component={RepoDetail} />
				<Route path="/userDetail/__/auth/" component={AuthComp} />
			</Switch>
			<Footer />
		</>
	);
}

export default App;
