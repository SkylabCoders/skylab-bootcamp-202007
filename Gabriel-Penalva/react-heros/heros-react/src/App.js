import React from 'react';
import DashBoard from './components/DashBoard';
import HeroList from './components/HeroList';
import Header from './components/Header-heros';
import Detail from './components/Details';
import { Route, Switch, BrowserRouter as Router } from "react-router-dom";
import heroList from './hero.mock'
import PageNotFound from './NotFound'


function App(props) {

    return (
        <Router>
            <div>
                <Header />
                <Switch>
                    <Route path='/' exact > <DashBoard heroList={heroList} /></Route>
                    <Route path='/herolist' component={() => <HeroList heroList={heroList} />} />
                    <Route path='/id/:heroId' component={(props) => <Detail heroList={heroList} {...props} />} />
                    <Route component={PageNotFound} />
                </Switch>

            </div>
        </Router>
    );
}

export default App;