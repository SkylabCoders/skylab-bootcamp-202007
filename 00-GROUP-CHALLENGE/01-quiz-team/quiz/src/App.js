import React from 'react';
import './css/App.css';
import { Route, Switch } from 'react-router-dom';
import PageNotFound from './components/PageNotFound'
import Home from './components/Home'
import Header from './components/Header'
import Footer from './components/Footer'
import LeaderBoard from './components/LeaderBoard'
import LoginScreen from './components/LoginScreen'
import Dashboard from './components/Dashboard'


function App() {
	return (
		<div className="app">
			<Header />
			<Switch>
				<Route path="/" exact component={Dashboard} />
				<Route path="/leaderboard" exact component={LeaderBoard} />
				<Route path="/login" exact component={LoginScreen} />
				<Route component={PageNotFound} />
			</Switch>
			<Footer />
		</div>
	);
}

export default App;
