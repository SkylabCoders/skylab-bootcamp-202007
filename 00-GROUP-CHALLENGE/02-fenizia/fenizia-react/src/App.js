import React from "react";
import { Route, Switch } from "react-router-dom";
import "./App.css";
import Header from "./components/Header";
import Dashboard from "./components/Dashboard";
import Footer from "./components/Footer";
import Profile from "./components/Profile";
import Login from "./components/Login";
import ProductList from "./components/ProductList";
import Cart from "./components/Cart";
import Register from "./components/Register";
import About from "./components/About";


function App(props) {
  return (
    <div className="container">
      <Header />
      
      <div className="margin-body">
        <Switch>
          <Route path="/" exact component={Dashboard} />
          <Route path="/profile" exact component={Profile} />
          <Route path="/register" exact component={Register} />
          <Route path="/cart" exact component={Cart} />
          <Route path="/login" exact component={Login} />
          <Route path="/cart" exact component={Cart} />
          <Route path="/register" exact component={Register} />
          <Route path="/sobre-nosotros" exact component={About} />
          <Route path="/:genre" exact component={ProductList} />
          <Route path="/infantil" exact component={ProductList} />
          <Route path="/romantico" exact component={ProductList} />
          <Route path="/historia" exact component={ProductList} />
          <Route path="/cocina" exact component={ProductList} />
          <Route path="/ensayo" exact component={ProductList} />
          <Route path="/thriller" exact component={ProductList} />
          <Route path="/todos" exact component={ProductList} />
          <Route path="/novedades" exact component={ProductList} />
          <Route path="/superventas" exact component={ProductList} />

          {/* <Route component={PageNotFound} /> */}
        </Switch>
      </div>
    
      <Footer />
    </div>
  );
}

export default App;
