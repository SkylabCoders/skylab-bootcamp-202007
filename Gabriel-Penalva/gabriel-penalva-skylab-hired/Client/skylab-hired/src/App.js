
import React, { useState, useEffect } from 'react';
import './App.css';
import Entry from './components/entry.component';
import Home from './components/home.component'
import { Route, Switch } from "react-router-dom";
import Nav from './components/nav.component';
import Logo from './components/logo.component'
import Login from './components/login.component';
import Registrer from './components/register.component';
import UserEntries from './components/userEntries.component'
import Profile from './components/profile.component'
import Footer from './components/footer.component'
import EntryComment from './components/entryComment.component'
import { Fragment } from 'react';
import store from './store/store'
import { loadDetails } from './actions/userActions';
import AddEntry from './components/addEntry.component';

function App() {
  const [message, setMessage] = useState(false);
  const [user, setUser] = useState(null);
  const [details, setDetails] = useState(null);

  useEffect(() => {
    store.addChangeListener(onChange);
    let usu = store.getUser();
    if (!details) loadDetails()
    if (!user) {
      setUser(usu);
    }
    return () => {
      return store.removeChangeListener(onChange);
    };
  }, [user, details]);

  function onChange() {
    setUser(store.getUser())
    setMessage(store.getMessage())
    setDetails(store.getDetail())

  }

  return (
    <div>

      {!user && (
        <Switch>
          <Route path='/' exact render={props =>
            <Fragment>
              <Logo />
              <Login />
              {!!message && message}
              <Footer />
            </Fragment>}
          />
          <Route path='/register' render={props =>
            <Fragment>
              <Logo />
              <Registrer />
              {!!message && message}
              <Footer />
            </Fragment>}
          />
        </Switch>)}

      {!!user && (

        <Switch>
          <Route path='/' exact render={props =>
            <Fragment>
              <Logo />
              <Home />
              {!!message && message}
              <Footer />
            </Fragment>
          } />


          <Route path='/list' render={props =>
            <Fragment>
              <Nav />
              {!!details && <Entry list={details} />}
              {!!message && message}
              <Footer />
            </Fragment>
          } />
          <Route path='/profile' render={props =>
            <Fragment>
              <Nav />
              <Profile />
              {!!message && message}
              <Footer />
            </Fragment>
          } />
          <Route path='/user/favoruites' render={props =>
            <Fragment>
              <Nav />
              <UserEntries type={'favourites'} />
              {!!message && message}
              <Footer />
            </Fragment>
          } />
          <Route path='/user/entries' render={props =>
            <Fragment>
              <Nav />
              <UserEntries type={'entries'} />
              {!!message && message}
              <Footer />

            </Fragment>
          } />

          <Route path='/comments/:entryId' render={props =>
            <Fragment>
              <Nav />
              <EntryComment props={props} />
              {!!message && message}
              <Footer />
            </Fragment>
          } />
          <Route path='/addEntry' render={props =>
            <Fragment>
              <Nav />
              <AddEntry />
              {!!message && message}
              <Footer />
            </Fragment>
          } />


        </Switch>)
      }
    </div >
  );
}

export default App;
