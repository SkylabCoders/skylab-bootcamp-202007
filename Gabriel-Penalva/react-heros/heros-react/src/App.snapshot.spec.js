import renderer from 'react-test-renderer';
import React from 'react';
import App from './App';
const tree = renderer.create(
    <App />
)

describe('App Snapshot', function () {
    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    })

});