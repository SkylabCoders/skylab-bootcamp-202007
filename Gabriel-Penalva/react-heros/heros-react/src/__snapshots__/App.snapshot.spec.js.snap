// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot should match 1`] = `
<div>
  <div
    className="header"
  >
    <h1>
      Tour of Heroes
    </h1>
    <nav>
      <a
        aria-current="page"
        className="my-active-class"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Dashboard
      </a>
       
      <a
        aria-current={null}
        href="/herolist"
        onClick={[Function]}
      >
        Heroes List
      </a>
    </nav>
  </div>
   
  <div
    className="center "
  >
    <h3>
      Top Heros:
    </h3>
    <div
      className="modules"
    >
      <div
        className="module"
      >
        <a
          className="dashlist hero-list_item"
          href="/id/11"
        >
            
          Dr Nice
        </a>
      </div>
      <div
        className="module"
      >
        <a
          className="dashlist hero-list_item"
          href="/id/12"
        >
            
          Narco
        </a>
      </div>
      <div
        className="module"
      >
        <a
          className="dashlist hero-list_item"
          href="/id/13"
        >
            
          Bombasto
        </a>
      </div>
      <div
        className="module"
      >
        <a
          className="dashlist hero-list_item"
          href="/id/14"
        >
            
          Celeritas
        </a>
      </div>
    </div>
  </div>
</div>
`;
